snow/validators/state.go

```diff
@@ -31,7 +31,7 @@ type State interface {
 		ctx context.Context,
 		height uint64,
 		subnetID ids.ID,
-	) (map[ids.NodeID]*GetValidatorOutput, error)
+	) ([]*GetValidatorOutput, error)
```

Comment:
I doubt that we have a single, canonical way to order validators, as returning a list seems to suggest.
warp package has its own order; proposerVM has its own order (and canonical too, to make sure each node order validators the same way; platformVM stakers implementation has its own ways.
I guess, if we want to return a list, we should pass a comparison function to signal what order client wants?